packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +119,67 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const {
+			hasInserterItems,
+			__experimentalHasOnlyOneAllowedBlockType,
+			__experimentalGetTheOnlyAllowedBlockType,
+		} = select( 'core/block-editor' );
+		const allowedBlock = getBlockType( __experimentalGetTheOnlyAllowedBlockType( rootClientId ) );
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOnlyOneAllowedInserterItem: __experimentalHasOnlyOneAllowedBlockType( rootClientId ),
+			blockTitle: allowedBlock ? allowedBlock.title : '',
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		return {
+			insertTheOnlyAllowedItem: () => {
```

Comment:
I think 'The' should be dropped from the function name. Probably call it a block instead of an item. Could also potentially use the object shorthand too for minor gains 😄 :
```
insertOnlyAllowedBlock() {
```